docs/dev/development-process/development-process.rst

```diff
@@ -0,0 +1,220 @@
+    </div>
+
+To facilitate this process, we have introduced UICompanion, a tool that integrates seamlessly with GitHub and Figma to assist developers through the UI/UX design phase.
+
+What is UICompanion?
+--------------------
+UICompanion is a combination of a GitHub bot and a Figma plugin designed to support developers working on the Artemis project through the UI/UX design step. This tool automates several aspects of the design process, making it easier for developers to create and iterate on UI designs.
+
+How to Use UICompanion
+----------------------
+When an issue is labeled with "UI" on GitHub, UICompanion automatically attaches a comment to the issue prompting the developer to start the UI design process in Figma. Here's how to proceed:
+
+1. **Initiate Design Process**: Upon receiving the UICompanion comment, click on the provided Figma Project Link to access our Figma project.
+2. **Follow Guidance**: Inside Figma, use the UICompanion Figma app for step-by-step guidance through the design process. This includes creating prototypes and seeking approval from designated Artemis maintainers.
+3. **Automate Tasks**: UICompanion helps automate tasks such as adding the necessary UI components from the Artemis Design System, ensuring that designs remain consistent with the project's UI guidelines.
+
+For more details and to get started with UICompanion, visit the `UICompanion Repository on GitHub <https://github.com/ls1intum/UICompanion>`_.
+
+4. Implement the Feature
+========================
+In this step, the development team converts the detailed plans and designs outlined in the functional proposal into working code. 
+This step requires careful adherence to the previously defined requirements and system architecture to ensure that the function fits seamlessly into the existing system and fulfills the specified functional and performance criteria.
+
+.. note::
+      Make sure to follow the `Artemis Code and Design Guidelines <https://docs.artemis.cit.tum.de/dev/guidelines.html>`_.
+
+5. Create a Pull Request
+========================
+After the feature implementation is complete, the developer is required to create a pull request for integrating the feature into the develop branch. 
+The subsequent sections provide guidance on the naming conventions and outline the necessary steps for creating and merging a pull request.
+
+Naming Conventions for GitHub Pull Requests
+-------------------------------------------
+
+1. The first term is a main feature of Artemis and should use code highlighting, e.g.  “``Programming exercises``:”.
+
+    1. Possible feature tags are: ``Programming exercises``, ``Quiz exercises``, ``Modeling exercises``, ``Text exercises``, ``File upload exercises``, ``Exam mode``,
+       ``Grading``, ``Assessment``, ``Communication``, ``Notifications``, ``Team exercises``, ``Lectures``, ``Plagiarism checks``, ``Learning analytics``,
+       ``Adaptive learning``, ``Learning path``, ``Tutorial groups``, ``Iris``.
```

Comment:
Make sure this list is complete. It is missing at least "Integrated code lifecyle"